src/drivers/vga/text_mode.rs

```diff
@@ -101,10 +83,15 @@ impl Writer {
 
         self.handle_scrolling();
 
+        let color_char = self.state.ransid_process(ch);
+        if color_char.ascii == b'\0' {
+            return 0;
+        }
+
         // Actually write the character to the screen, escapes have been handled
         // previously no need to worry about those anymore.
         let pos: u16 = (self.y * WIDTH + self.x) as u16;
-        let byte: u16 = ((Color::LightGreen as u16) << 8) | u16::from(ch);
+        let byte: u16 = ((color_char.style as u16) << 8) | u16::from(color_char.ascii);
```

Comment:
Don't think the `: u16` type annotation is needed here

Reply:
it is

Reply:
kk